components/match2/commons/match_util.lua

```diff
@@ -0,0 +1,27 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Match/Util
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Table = require('Module:Table')
+
+local MatchUtil = {}
+
+---@param matchOpponent table
+---@param gameOpponent table
+function MatchUtil.enrichGameOpponentFromMatchOpponent(matchOpponent, gameOpponent)
+	local newGameOpponent = Table.deepMerge(matchOpponent, gameOpponent)
+	-- These values are only allowed to come from Game and not Match
+	newGameOpponent.placement = gameOpponent.placement
+	newGameOpponent.score = gameOpponent.score
+	newGameOpponent.status = gameOpponent.status
+
+	-- TODO: match2players vs players duplication. Which one to keep? How to merge?
```

Comment:
mind in ms you pass a standardOpponent object in here, while in match ticker you pass a lpdb_match2opponent object

so for the ms call the deep merge already takes care of e.g. the faction issue
while for match ticker it remains unsolved